src/app/modules/item/components/thread/thread.component.ts

```diff
@@ -11,25 +10,45 @@ import { ThreadService } from '../../services/threads.service';
   styleUrls: [ './thread.component.scss' ],
 })
 export class ThreadComponent implements OnChanges, OnDestroy {
-  @Input() itemData?: ItemData;
+  @Input() itemId?: string;
 
-  messages$ = this.threadService.messages$;
-  threadClosed$ = this.messages$.pipe(map(messages => messages.some(message => message.eventType === 'thread_closed')));
+  messageToSend = '';
+  openingThread = false;
+  widgetOpened = false;
 
-  private itemData$ = new ReplaySubject<ItemData>(1);
+  events$ = this.threadService.events$;
+  threadStatus$ = this.threadService.status$;
+  canOpenThread$ = combineLatest([ this.groupWatchingService.isWatching$, this.threadStatus$ ]).pipe(
+    map(([ isWatching, status ]) => !isWatching && status !== 'opened'),
+  );
+  canCloseThread$ = combineLatest([ this.groupWatchingService.isWatching$, this.threadStatus$ ]).pipe(
+    map(([ isWatching, status ]) => !isWatching && status === 'opened'),
+  );
+
+  private itemId$ = new ReplaySubject<string>(1);
 
   private subscriptions = [
-    combineLatest([ this.itemData$, this.userSession.userProfile$, this.groupWatchingService.watchedGroup$ ])
+    combineLatest([ this.itemId$, this.userSession.userProfile$, this.groupWatchingService.watchedGroup$ ])
       .pipe(catchError(() => EMPTY)) // error is handled elsewhere
-      .subscribe(([ itemData, profile, watchedGroup ]) => {
+      .subscribe(([ itemId, profile, watchedGroup ]) => {
         this.threadService.init({
           participantId: watchedGroup?.route.id || profile.groupId,
-          itemId: itemData.item.id,
+          itemId,
           userId: profile.groupId,
           isMine: !watchedGroup,
           canWatchParticipant: !!watchedGroup,
         });
       }),
+    this.itemId$
+      .pipe(switchMap(() => this.threadStatus$), take(1))
+      .subscribe(status => {
+        this.threadService.follow();
+        this.widgetOpened = status === 'opened';
+      }),
+
+    this.itemId$.pipe(pairwise()).subscribe(() => this.threadService.unfollow()),
```

Comment:
You use pairwise jut to skip the first event? 🤨 Just use `skip(1)`.

Reply:
Indeed :monocle_face:, fixed